src/db-up/azext_db_up/_validators.py

```diff
@@ -30,43 +30,59 @@ def db_down_namespace_processor(db_type):
     return lambda cmd, namespace: _process_db_down_namespace(namespace, db_type=db_type)
 
 
+# pylint: disable=bare-except
 def _process_db_up_namespace(cmd, namespace, db_type=None):
-    # populate from cache if existing
-    _set_value(db_type, namespace, 'location', 'location', cache=False)
-    _set_value(db_type, namespace, 'resource_group_name', 'group', cache=False)
+    # populate from cache if existing when no resource group name provided
+    resource_client = resource_client_factory(cmd.cli_ctx)
+    if namespace.resource_group_name is None:
+        _set_value(db_type, namespace, 'resource_group_name', 'group', cache=False)
+        try:
+            resource_client.resource_groups.get(namespace.resource_group_name)
+        except:  # Clear resource group name information when it is invalid
+            namespace.resource_group_name = None
 
-    # generate smart defaults
+    # populate from cache if existing when no location provided
+    if namespace.location is None:
+        _set_value(db_type, namespace, 'location', 'location', cache=False)
+    # generate smart defaults when namespace.location is None
     if _get_value(db_type, namespace, 'location', 'location') is None:
         try:
             get_default_location_from_resource_group(cmd, namespace)
         except (CLIError, ValidationError):
-            namespace.location = 'westus2'
-    _set_value(db_type, namespace, 'location', 'location', namespace.location)
+            namespace.location = 'eastus'
+    _set_value(db_type, namespace, 'location', 'location', default=namespace.location)
```

Comment:
Why is it changed from `westus2` to `eastus`

Reply:
westus2 will be not available in some cases.